src/Akeneo/Platform/Bundle/ImportExportBundle/Query/GetJobExecutionTracking.php

```diff
@@ -0,0 +1,180 @@
+        if (!$jobExecution instanceof JobExecution) {
+            throw new \Exception(); //@TODO create execution + test
+        }
+
+        $jobExecution = $this->jobExecutionManager->resolveJobExecutionStatus($jobExecution);
+        $jobName = $jobExecution->getJobInstance()->getJobName();
+        $stepExecutions = $jobExecution->getStepExecutions();
+
+        /* What do we do if we have a UndefinedJobException ? */
+        $job = $this->jobRegistry->get($jobName);
+
+        $jobExecutionTracking = new JobExecutionTracking();
+        $jobExecutionTracking->status = $this->getMappedStatus($jobExecution->getStatus());
+        $jobExecutionTracking->currentStep = count($jobExecution->getStepExecutions());
+        $jobExecutionTracking->totalSteps = count($job->getSteps());
+        $jobExecutionTracking->steps = $this->getStepExecutionTracking($job, $stepExecutions);
+
+        return $jobExecutionTracking;
+    }
+
+    private function getStepExecutionTracking(JobInterface $job, $stepExecutions): array
+    {
+        $stepsExecutionTracking = [];
+
+        /** @TODO add JobWithStepsInterface to assert getSteps function exist */
+        /** @var StepInterface $step */
+        foreach ($job->getSteps() as $step) {
+            $stepName = $step->getName();
+            $stepExecutionIndex = $this->searchFirstMatchingStepExecutionIndex($stepExecutions, $stepName);
+            if ($stepExecutionIndex === -1) {
+                $stepsExecutionTracking[] = $this->createStepExecutionTrackingNotStartedFromStep($step);
+                continue;
+            }
+
+            $stepExecution = $stepExecutions[$stepExecutionIndex];
+            $stepsExecutionTracking[] = $this->createStepExecutionTrackingFromStepAndStepExecution($step, $stepExecution);
+            unset($stepExecutions[$stepExecutionIndex]);
+        }
+
+        return $stepsExecutionTracking;
+    }
+
+    private function searchFirstMatchingStepExecutionIndex(PersistentCollection $stepExecutions, string $stepName)
+    {
+        foreach ($stepExecutions as $stepExecutionIndex => $stepExecution) {
+            if ($stepExecution->getStepName() === $stepName) {
+                return $stepExecutionIndex;
+            }
+        }
+
+        /** @TODO maybe use Exception */
+        return -1;
+    }
+
+    private function createStepExecutionTrackingNotStartedFromStep(StepInterface $step)
+    {
+        $stepExecutionTracking = new StepExecutionTracking();
+        $stepExecutionTracking->name = $step->getName();
+        $stepExecutionTracking->status = 'NOT STARTED';
+        if ($step instanceof TrackableStepInterface && $step->isTrackable()) {
+            $stepExecutionTracking->isTrackable = true;
+        }
+
+        return $stepExecutionTracking;
+    }
+
+    private function createStepExecutionTrackingFromStepAndStepExecution(StepInterface $step, StepExecution $stepExecution)
+    {
+        $duration = $this->calculateDuration($stepExecution);
```

Comment:
calculate doesn't feel right

```suggestion
        $duration = $this->computeDuration($stepExecution);
```